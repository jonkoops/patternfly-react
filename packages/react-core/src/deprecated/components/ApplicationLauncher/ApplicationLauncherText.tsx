import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/AppLauncher/app-launcher';

export interface ApplicationLauncherTextProps extends HTMLProps<HTMLSpanElement> {
  /** content rendered inside the text container */
  children: ReactNode;
  /** Additional classes added to the text container */
  className?: string;
}

export const ApplicationLauncherText: FunctionComponent<ApplicationLauncherTextProps> = ({
  className = '',
  children,
  ...props
}: ApplicationLauncherTextProps) => (
  <span className={css(`${styles.appLauncherMenuItem}-text`, className)} {...props}>
    {children}
  </span>
);
ApplicationLauncherText.displayName = 'ApplicationLauncherText';
