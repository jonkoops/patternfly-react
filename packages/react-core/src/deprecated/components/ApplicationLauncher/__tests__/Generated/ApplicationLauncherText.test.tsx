/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';

import { ApplicationLauncherText } from '../../ApplicationLauncherText';

it('ApplicationLauncherText should match snapshot (auto-generated)', () => {
  const { asFragment } = render(<ApplicationLauncherText children={<div>ReactNode</div>} className={"''"} />);
  expect(asFragment()).toMatchSnapshot();
});
