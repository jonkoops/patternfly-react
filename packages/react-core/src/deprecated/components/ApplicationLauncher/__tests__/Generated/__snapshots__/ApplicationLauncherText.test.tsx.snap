// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ApplicationLauncherText should match snapshot (auto-generated) 1`] = `
<DocumentFragment>
  <span
    class="pf-v5-c-app-launcher__menu-item-text ''"
  >
    <div>
      ReactNode
    </div>
  </span>
</DocumentFragment>
`;
